### PR TITLE
Free getline buffer after use

### PR DESCRIPTION
**Pull request by HJLeee**
Fill in `free()` for `getline()` buffer (Reported by static analyzer)

### CODE REVIEW

File: src/coreclr/pal/src/thread/process.cpp [hunk 1]
```diff
@@ -1675,6 +1675,7 @@ GetProcessIdDisambiguationKey(DWORD processId, UINT64 *disambiguationKey)
     {
         TRACE("GetProcessIdDisambiguationKey: getline() FAILED");
         SetLastError(ERROR_INVALID_HANDLE);
+        free(line);
```
Comment (stephentoub) [inline src/coreclr/pal/src/thread/process.cpp:1678]: getline allocates a buffer and returns that allocated buffer even if it fails?

Comment (HJLeee) [reply src/coreclr/pal/src/thread/process.cpp:1678]: Yes, the `getline()` doc says as below; 
https://man7.org/linux/man-pages/man3/getline.3.html
```
       If *lineptr is set to NULL before the call, then getline() will
       allocate a buffer for storing the line.  This buffer should be
       freed by the user program even if getline() failed.
```

Comment (HJLeee) [reply src/coreclr/pal/src/thread/process.cpp:1678]: Code reference
https://github.com/lattera/glibc/blob/895ef79e04a953cac1493863bcae29ad85657ee1/stdio-common/getline.c#L28
https://github.com/lattera/glibc/blob/895ef79e04a953cac1493863bcae29ad85657ee1/libio/iogetdelim.c#L59-L62

Comment (stephentoub) [reply src/coreclr/pal/src/thread/process.cpp:1678]: Does this also need to fclose(statFile)?

File: src/coreclr/pal/src/thread/process.cpp [hunk 2]
```diff
@@ -1675,6 +1675,7 @@ GetProcessIdDisambiguationKey(DWORD processId, UINT64 *disambiguationKey)
...
         return FALSE;
     }
```
Comment (stephentoub) [inline src/coreclr/pal/src/thread/process.cpp:1681]: It looks like https://github.com/dotnet/runtime/pull/74899/files#diff-236bca3ea8ce55b54c55c47ab9d21a4db9596eb3fbc6052565a81a2ebb3ee73fR1701-R1702 should also be moved up to before the `if (sscanfRet != 1)` check?

File: src/coreclr/pal/src/thread/process.cpp (hunk 1 above)
Comment (HJLeee) [reply src/coreclr/pal/src/thread/process.cpp:1678]: Fixed. Thanks.

File: src/coreclr/pal/src/thread/process.cpp (hunk 2 above)
Comment (HJLeee) [reply src/coreclr/pal/src/thread/process.cpp:1682]: Right. Fixed. :)

**Comment by HJLeee** [comment]
@stephentoub Would you review this please?

**Review by stephentoub — APPROVED** [review]

**Review by janvorli — APPROVED** [review]
LGTM, thank you

**Pull request merged**